js-core/lesson1.md

```diff
@@ -4,198 +4,428 @@
+If you played around with `console.log()` in the last step, you might have noticed that words always have to be inside quotes or else the code breaks, whereas numbers can work without quotes. Why is this? Well these are different **data types** in JavaScript. You'll need to learn the three main data types to get started coding (later we'll look at some other data types, like objects and arrays).
+
+## Strings
+
+Strings are simple text, ranging anywhere from a single character to lengthy paragraphs. A string is always written inside a pair of quotes (single or double).
 
 ```js
-// Arrays can hold any data type
-var countries = ['Scotland', 'Germany', 'Syria'];
-var numbers = [7, 1, 23, 19, 11, 5, 16, 31];
+'node girls'
+```
+
+## Numbers
 
-// Data types can be mixed, but be wary of that!
-var mixed = ['Tim', 27];
+Numbers do not need to be wrapped in quotes like strings, and can just be written as is.
 
-// Arrays can also be empty
-var emptyArray = [];
+```js
+100 // an integer number
+100.89 // a decimal number (floating point)
 ```
 
-### Reading, writing and length
+## Booleans
+
+There is a special data type in JavaScript known as a "boolean" value. A boolean is either `true` or `false`, and it should be written without quotes.
+
+```js
+true
+```
+
+
+You've seen what they look like, now make sure they are what you think they are. There is a built-in feature in JavaScript which allows you to check the type of a particular value: `typeof`.
+
+Write this code in your `main.js`, and run it (using `node main.js` in the terminal):
+```js
+console.log(typeof 'Hello world!');
+```
+You should see `string` printed to the console.
+
+> **Exercise** Now try that three times, once for each data type. Log the `typeof` for three different values: `"Code Your Future"`, `6` and `false`. You should get different output for each one.
+
+# Variables
+
+When you code, you'll want to create *shortcuts* so you can easily refer to things without writing out the same value every time. In JavaScript we have **variables**, which allow us to store values to use later.
+
+> Think of variables names like **labels** on boxes, while the value of the variable are the **contents** of the box - you could change the contents of a box and leave the label intact, the contents of the boxes can have different types, the boxes should have good labels (a box of books being labeled pens would be very confusing),
+>
+> ![Variables are like boxes](http://2.bp.blogspot.com/-guGunS9ljms/Ui-sF7Q9yQI/AAAAAAAAACU/DdnRgAARYyI/s1600/box.png)
+> Photo from [Khan Academy](http://cs-blog.khanacademy.org/2013/09/teaching-variables-analogies-and.html)
+
+## Try it out
+
+You create a variable using the keyword `var`, which lets the application know that you're about to store a value. You also have to give a name to your variable, which can be anything you like, so long as it doesn't include spaces.
```

Comment:
They won't hit these problems at lesson 1, and the majority of examples online would use var, I think it would be confusing if we just skipped over var immediately.

Could we introduce const and let straight away in lesson 2 and encourage them to use them from that point on? (and if so, how do we ease the confusion about the existence of two "versions" of JS)